### PR TITLE
Fs module v0.0.2

### PR DESCRIPTION
**Pull request by cortexlfj**
Add New Features:
1.  PauseLocalSeed
2.  ResumeLocalSeed
3.  ListAllTorrents

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -115,6 +118,101 @@ type TorrentManager struct {
...
+		tm.localSeedFiles[ih] = resumeFlag
+	}
+
+	return nil
+}
+
+// divide localSeed/on-chain Files
+// return status of torrents
+func (tm *TorrentManager) listAllTorrents() map[string]map[string]int {
+	tm.lock.RLock()
```
Comment (ucwong) [inline handler.go:199]: Is ```tm.lock.RLock()``` neccessary ?  if only ```tm.localSeedFiles``` should to be locked

File: fs.go [hunk 2]
```diff
@@ -399,8 +399,8 @@ func (fs *TorrentFS) SeedingLocal(ctx context.Context, filePath string, isLinkMo
 	// 4. copy or link, will not cover if dst exist!
 	ih := common.Address(mi.HashInfoBytes())
 	log.Info("Local file Seeding", "ih", ih.Hex(), "path", dataPath)
-	linkDst := strings.TrimPrefix(strings.ToLower(ih.Hex()), common.Prefix)
-	linkDst = filepath.Join(fs.storage().TmpDataDir, linkDst)
+	ihHexLower := strings.TrimPrefix(strings.ToLower(ih.Hex()), common.Prefix)
```
Comment (ucwong) [inline fs.go:402]: ```infoHash = strings.TrimPrefix(strings.ToLower(ih.Hex()), common.Prefix)``` is it OK to set ih directly here ?

File: handler.go [hunk 3]
```diff
@@ -115,6 +118,101 @@ type TorrentManager struct {
...
+// can only call by fs.go: 'SeedingLocal()'
+func (tm *TorrentManager) addLocalSeedFile(ih string) bool {
+	if !common.IsHexAddress(ih) {
+		return false
+	}
+	ih = strings.TrimPrefix(strings.ToLower(ih), common.Prefix)
+
+	tm.localSeedLock.Lock()
+	defer tm.localSeedLock.Unlock()
+
```
Comment (ucwong) [inline handler.go:136]: Move ```tm.localSeedLock.Lock()
	defer tm.localSeedLock.Unlock()``` 2 lines down below ```GoodFiles``` validation for ``` GoodFiles``` is static

File: handler.go [hunk 4]
```diff
@@ -115,6 +118,101 @@ type TorrentManager struct {
...
+	} else if !valid {
+		return errors.New(fmt.Sprintf("Local Seeding File Is Not Seeding<%s>", ih))
+	}
+
+	if t := tm.getTorrent(ih); t != nil {
+		log.Debug("TorrentFS", "from seed to pause", "ok")
+		t.Pause()
+	}
+	tm.localSeedFiles[ih] = false
+
```
Comment (ucwong) [inline handler.go:166]: ```tm.localSeedFiles[ih] = !t.Paused()``` I think the torrent status will be paused certainly , but get the torrent current status  for checking out will be more graceful.

File: fs.go (hunk 2 above)
Comment (cortexlfj) [reply fs.go:402]: Definitely, it's an unnecessary variable initialize here.

File: handler.go (hunk 3 above)
Comment (cortexlfj) [reply handler.go:136]: Got

**Review by ucwong — APPROVED** [review]
LGTM

**Pull request merged**